backend-api/src/functions/asyncFinishBiometricSession/asyncFinishBiometricSessionHandler.ts

```diff
@@ -23,20 +35,176 @@ export async function lambdaHandlerConstructor(
   logger.resetKeys();
   logger.addContext(context);
   logger.info(LogMessage.FINISH_BIOMETRIC_SESSION_STARTED);
+  const configResult = getFinishBiometricSessionConfig(dependencies.env);
+  if (configResult.isError) {
+    return serverErrorResponse;
+  }
+
+  const config = configResult.value;
 
-  const validateRequestBodyResult = validateRequestBody(event.body);
-  if (validateRequestBodyResult.isError) {
-    const errorMessage = validateRequestBodyResult.value.errorMessage;
+  const validateResult = validateRequestBody(event.body);
+
+  if (validateResult.isError) {
     logger.error(LogMessage.FINISH_BIOMETRIC_SESSION_REQUEST_BODY_INVALID, {
-      errorMessage,
+      errorMessage: validateResult.value.errorMessage,
     });
-    return badRequestResponse("invalid_request", errorMessage);
+    return badRequestResponse(
+      "invalid_request",
+      validateResult.value.errorMessage,
+    );
+  }
+
+  const { sessionId, biometricSessionId } = validateResult.value;
+  const eventService = dependencies.getEventService(config.TXMA_SQS);
+  const sessionRegistry = dependencies.getSessionRegistry(
+    config.SESSION_TABLE_NAME,
+  );
+
+  const updateResult = await sessionRegistry.updateSession(
+    sessionId,
+    new BiometricSessionFinished(biometricSessionId),
+  );
+
+  if (updateResult.isError) {
+    return handleUpdateSessionError(
+      updateResult,
+      eventService,
+      sessionId,
+      biometricSessionId,
+      config.ISSUER,
+    );
   }
 
   logger.info(LogMessage.FINISH_BIOMETRIC_SESSION_COMPLETED);
   return notImplementedResponse;
 }
 
+async function handleConditionalCheckFailure(
+  eventService: IEventService,
+  sessionAttributes: SessionAttributes,
+  biometricSessionId: string,
+  issuer: string,
+): Promise<APIGatewayProxyResult> {
+  const sessionAge = Date.now() - sessionAttributes.createdAt;
+  const isSessionExpired = sessionAge > 60 * 60 * 1000;
+
+  const writeEventResult = await eventService.writeGenericEvent({
+    eventName: "DCMAW_ASYNC_CRI_4XXERROR",
+    sub: sessionAttributes.subjectIdentifier,
+    sessionId: sessionAttributes.sessionId,
+    govukSigninJourneyId: sessionAttributes.govukSigninJourneyId,
+    componentId: issuer,
+    getNowInMilliseconds: Date.now,
+    transactionId: biometricSessionId,
+    extensions: isSessionExpired
+      ? { suspected_fraud_signal: "AUTH_SESSION_TOO_OLD" }
+      : undefined,
```

Comment:
nitpick (non-blocking): we could have a helper function for this block

Originally Chris provided this feedback to me for the restricted blocks in the eventService but more because it reduced duplication I think. This wouldn't reduce duplication here but it would remove the cumbersome expression

```suggestion
    extensions: getFraudSignals(isSessionExpired)
```

Happy for you to resolve tho if you don't feel there's value